genesyscloud/architect_schedules/genesyscloud_architect_schedules_proxy.go

```diff
@@ -0,0 +1,205 @@
+*/
+
+// internalProxy holds a proxy instance that can be used throughout the package
+var internalProxy *architectSchedulesProxy
+
+// Type definitions for each func on our proxy so we can easily mock them out later
+type createArchitectSchedulesFunc func(ctx context.Context, p *architectSchedulesProxy, schedules *platformclientv2.Schedule) (*platformclientv2.Schedule, *platformclientv2.APIResponse, error)
+type getAllArchitectSchedulesFunc func(ctx context.Context, p *architectSchedulesProxy) (*[]platformclientv2.Schedule, *platformclientv2.APIResponse, error)
+type getArchitectSchedulesIdByNameFunc func(ctx context.Context, p *architectSchedulesProxy, name string) (id string, retryable bool, response *platformclientv2.APIResponse, err error)
+type getArchitectSchedulesByIdFunc func(ctx context.Context, p *architectSchedulesProxy, id string) (schedules *platformclientv2.Schedule, response *platformclientv2.APIResponse, err error)
+type updateArchitectSchedulesFunc func(ctx context.Context, p *architectSchedulesProxy, id string, schedules *platformclientv2.Schedule) (*platformclientv2.Schedule, *platformclientv2.APIResponse, error)
+type deleteArchitectSchedulesFunc func(ctx context.Context, p *architectSchedulesProxy, id string) (*platformclientv2.APIResponse, error)
+
+/*
+The architectSchedulesProxy struct holds all the methods responsible for making calls to
+the Genesys Cloud APIs. This means that within this struct, you'll find all the functions designed
+to interact directly with the various features and services offered by Genesys Cloud,
+enabling this terraform provider software to perform tasks like retrieving data, updating information,
+or triggering actions within the Genesys Cloud environment.
+*/
+type architectSchedulesProxy struct {
+	clientConfig                      *platformclientv2.Configuration
+	architectApi                      *platformclientv2.ArchitectApi
+	createArchitectSchedulesAttr      createArchitectSchedulesFunc
+	getAllArchitectSchedulesAttr      getAllArchitectSchedulesFunc
+	getArchitectSchedulesIdByNameAttr getArchitectSchedulesIdByNameFunc
+	getArchitectSchedulesByIdAttr     getArchitectSchedulesByIdFunc
+	updateArchitectSchedulesAttr      updateArchitectSchedulesFunc
+	deleteArchitectSchedulesAttr      deleteArchitectSchedulesFunc
+}
+
+/*
+The function newArchitectSchedulesProxy sets up the architect schedules proxy by providing it
+with all the necessary information to communicate effectively with Genesys Cloud.
+This includes configuring the proxy with the required data and settings so that it can interact
+seamlessly with the Genesys Cloud platform.
+*/
+func newArchitectSchedulesProxy(clientConfig *platformclientv2.Configuration) *architectSchedulesProxy {
+	api := platformclientv2.NewArchitectApiWithConfig(clientConfig) // NewArchitectApiWithConfig creates an Genesyc Cloud API instance using the provided configuration
```

Comment:
We should include caching in this object.  @dginty4 or @charliecon can you chat with Suresh and walk him through how we have implemented caching for the getAll and get objects

Reply:
> We should include caching in this object. @dginty4 or @charliecon can you chat with Suresh and walk him through how we have implemented caching for the getAll and get objects

caching is added